src/ResourceManager/KeyVault/Commands.KeyVault/Commands/RestoreAzureKeyVaultKey.cs

```diff
@@ -24,42 +24,70 @@ namespace Microsoft.Azure.Commands.KeyVault
     /// </summary>
     [Cmdlet(VerbsData.Restore, "AzureKeyVaultKey",
         SupportsShouldProcess = true,
+        DefaultParameterSetName = ByVaultNameParameterSet,
         HelpUri = Constants.KeyVaultHelpUri)]
-    [OutputType(typeof(KeyBundle))]
+    [OutputType(typeof(PSKeyVaultKey))]
     public class RestoreAzureKeyVaultKey : KeyVaultCmdletBase
     {
+        #region Parameter Set Names
+
+        private const string ByVaultNameParameterSet = "ByVaultName";
+        private const string ByInputObjectParameterSet = "ByInputObject";
+
+        #endregion
+
         #region Input Parameter Definitions
 
         /// <summary>
         /// Vault name
         /// </summary>
         [Parameter(Mandatory = true,
                    Position = 0,
+                   ParameterSetName = ByVaultNameParameterSet,
                    ValueFromPipelineByPropertyName = true,
                    HelpMessage = "Vault name. Cmdlet constructs the FQDN of a vault based on the name and currently selected environment.")]
         [ValidateNotNullOrEmpty]
         public string VaultName { get; set; }
 
+        /// <summary>
+        /// KeyVault object
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   Position = 0,
+                   ParameterSetName = ByInputObjectParameterSet,
+                   ValueFromPipeline = true,
+                   HelpMessage = "KeyVault object")]
+        [ValidateNotNullOrEmpty]
+        public PSKeyVault InputObject { get; set; }
+
         /// <summary>
         /// The input file in which the backup blob is stored
         /// </summary>
         [Parameter(Mandatory = true,
                    Position = 1,
                    HelpMessage = "Input file. The input file containing the backed-up blob")]
         [ValidateNotNullOrEmpty]
-        public string InputFile { get; set; }
+        public string[] InputFile { get; set; }
```

Comment:
Why an array? Is there a scenario that requires the batching of multiple Restore cmdlets in a single one? I get the convenience factor, but I'm concerned that error scenarios will lead to confusion: what was restored, which one failed and why, and do we continue on error or stop at the first one?